File: template-method/java/XmlDataAnalyzer.java
package TemplateMethod;

// UML diyagramındaki AbstractClass (DataAnalyzerBase) sınıfından türer.
// UML diyagramındaki ConcreteClass yapısına denk gelmektedir.
class XmlDataAnalyzer extends DataAnalyzerBase {
    @Override
    public void exportResult(Object result) {
        // Durumun XML'e dönüştürülmesi işlemleri yer alacaktır.
        System.out.println("XML exported.");
    }
}

File: template-method/kotlin/XmlDataAnalyzer.kt
package TemplateMethod

// UML diyagramındaki AbstractClass (DataAnalyzerBase) sınıfından türer.
// UML diyagramındaki ConcreteClass yapısına denk gelmektedir.
internal class XmlDataAnalyzer : DataAnalyzerBase() {
    override fun exportResult(result: Any) {
        // Durumun XML'e dönüştürülmesi işlemleri yer alacaktır.
        println("XML exported.")
    }
}